src/dragonball/src/resource_manager.rs

```diff
@@ -0,0 +1,683 @@
+// Copyright (C) 2022 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Mutex;
+
+use dbs_allocator::{Constraint, IntervalTree, Range};
+use dbs_boot::layout::{
+    GUEST_MEM_END, GUEST_MEM_START, GUEST_PHYS_END, IRQ_BASE as LEGACY_IRQ_BASE,
+    IRQ_MAX as LEGACY_IRQ_MAX, MMIO_LOW_END, MMIO_LOW_START,
+};
+use dbs_device::resources::{DeviceResources, MsiIrqType, Resource, ResourceConstraint};
+
+// The LEGACY_IRQ_BASE irq is reserved for shared IRQ.
+const SHARED_IRQ: u32 = LEGACY_IRQ_BASE;
+#[cfg(target_arch = "x86_64")]
+const MSI_IRQ_BASE: u32 = 24;
+#[cfg(target_arch = "aarch64")]
```

Comment:
To avoid any confusion, can you add a `cfg` check so that if the target arch is not one of these two, the build fails with a message?

Could you also reference a GitHub issue (raise one if necessary) that lists the Kata architectures this code does not yet run on, along with some details of what would need to be done for those platforms to get dragonball working.

Reply:
Sure, we'll do that check in another PR. Issue #4316 is created for this.

Besides, the unsupported platforms issue is also created. #4320